File: sources/easygenerator.Web/App/models/video.js
define(['models/entity', 'constants'],
    function (Entity, constants) {
        "use strict";

        var video = function (spec) {

            var obj = new Entity(spec);
            obj.title = spec.title;
            obj.vimeoId = spec.vimeoId;
            obj.thumbnailUrl = spec.thumbnailUrl;
            obj.progress = spec.progress;
            obj.status = spec.status || constants.storage.video.statuses.loaded;
            obj.associatedLearningContentId = spec.associatedLearningContentId || null;
              
            return obj;
        };

        return video;
    }
);

File: sources/easygenerator.StorageServer/Models/Entities/Video.cs
using System;

namespace easygenerator.StorageServer.Models.Entities
{
    public class Video : Entity
    {
        protected internal Video() { }

        public string VimeoId { get; protected set; }
        public string Title { get; protected set; }
        public long Size { get; protected set; }
        public Guid UserId { get; protected set; }

        public Video(string title, long size, Guid userId)
        {
            Title = title;
            Size = size;
            UserId = userId;
        }

        public void SetVimeoId(string vimeoId)
        {
            VimeoId = vimeoId;
        }
    }
}